File: Xamarin_Forms/Xamarin_Forms/GertecXamarinForms/GertecXamarinForms.Android/TEF/MoneyTextWatcher.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.Icu.Math;
using Android.Icu.Text;
using Android.OS;
using Android.Runtime;
using Android.Text;
using Android.Views;
using Android.Widget;
using Java.Lang;
using Java.Util;

namespace GertecXamarinForms.Droid.TEF
{
    public class MoneyTextWatcher : Java.Lang.Object, ITextWatcher
    {
        private WeakReference<EditText> editTextWeakReference;
        private Locale locale;

        public MoneyTextWatcher(EditText editText, Locale locale)
        {
            this.editTextWeakReference = new WeakReference<EditText>(editText);
            this.locale = locale != null ? locale : Locale.Default;
        }

        public MoneyTextWatcher(EditText editText)
        {
            this.editTextWeakReference = new WeakReference<EditText>(editText);
            this.locale = Locale.Default;
        }

        public void AfterTextChanged(IEditable editable)
        {
            /*EditText editText = editTextWeakReference;
            if (editText == null) return;
            editText.RemoveTextChangedListener(this);

            BigDecimal parsed = parseToBigDecimal(editable.ToString(), locale);
            string formatted = NumberFormat.GetCurrencyInstance(locale).Format(parsed);
            // NumberFormat.getNumberInstance(locale).format(parsed); // sem o simbolo de moeda
            */

        }

        public void BeforeTextChanged(ICharSequence s, int start, int count, int after)
        {
           
        }

        public void OnTextChanged(ICharSequence s, int start, int before, int count)
        {
            
        }
        private BigDecimal parseToBigDecimal(string value, Locale locale)
        {
            string replaceable = string.Format("[%s,.\\s]", NumberFormat.GetCurrencyInstance(locale).Currency.Symbol);

            string cleanString = value.Replace(replaceable, "");

            return new BigDecimal(cleanString).SetScale(
                    2, (RoundOptions)Java.Math.RoundOptions.Floor).Divide(new BigDecimal(100), (RoundOptions)Java.Math.RoundOptions.Floor
            );
        }

    }
}

File: Android_Nativo_Java/Android/app/src/main/java/com/gertec/exemplosgertec/ExemploTEF/MoneyTextWatcher.java
package com.gertec.exemplosgertec.ExemploTEF;

import android.text.Editable;
import android.text.TextWatcher;
import android.widget.EditText;

import java.lang.ref.WeakReference;
import java.math.BigDecimal;
import java.text.NumberFormat;
import java.util.Locale;

public class MoneyTextWatcher implements TextWatcher {
    private final WeakReference<EditText> editTextWeakReference;

    public MoneyTextWatcher(EditText editText, Locale locale) {
        this.editTextWeakReference = new WeakReference<EditText>(editText);
    }

    public MoneyTextWatcher(EditText editText) {
        System.out.println(editText.getText().toString());
        this.editTextWeakReference = new WeakReference<EditText>(editText);
    }

    @Override
    public void beforeTextChanged(CharSequence s, int start, int count, int after) {

    }

    @Override
    public void onTextChanged(CharSequence s, int start, int before, int count) {

    }

    @Override
    public void afterTextChanged(Editable editable) {
        EditText editText = editTextWeakReference.get();
        if (editText == null) return;
        editText.removeTextChangedListener(this);
        BigDecimal parsed = parseToBigDecimal(editable.toString().replaceAll("R\\$", ""));
        String formatted = NumberFormat.getCurrencyInstance().format(parsed);
        // NumberFormat.getNumberInstance(locale).format(parsed); // sem o simbolo de moeda
        editText.setText(formatted.replaceAll("R\\$", ""));
        editText.setSelection(formatted.length()-2);
        editText.addTextChangedListener(this);
    }

    private BigDecimal parseToBigDecimal(String value) {
        String replaceable = String.format("[%s,.\\s]", NumberFormat.getCurrencyInstance().getCurrency().getSymbol());
        String cleanString = value.replaceAll(replaceable, "");
        return new BigDecimal(cleanString).setScale(
                2, BigDecimal.ROUND_FLOOR).divide(new BigDecimal(100), BigDecimal.ROUND_FLOOR
        );
    }
}
